app/src/org/commcare/android/resource/installers/ProfileAndroidInstaller.java

```diff
@@ -91,9 +97,11 @@ public boolean install(Resource r, ResourceLocation location, Reference ref, Res
                     upgrade ? Resource.RESOURCE_STATUS_UNINITIALIZED : Resource.RESOURCE_STATUS_UNINITIALIZED, false);
             
             Profile p = parser.parse();
+            instance.setProfile(p);
```

Comment:
[High importance] Question: Shouldn't this go into !upgrade below or stay in init? The profile isn't really "live" at this point in the chain, so this seems really dangerous to me.